### PR TITLE
Add TestIsWeakHashAlgo tests for is_weak_hash_algo() in ssl.py

### PR DESCRIPTION
**Pull request by Raavi29**
## Summary
Adds 11 unit tests for `is_weak_hash_algo()` in `nettacker/core/lib/ssl.py` as a new `TestIsWeakHashAlgo` class appended to the existing test suite.

## What was tested

- Weak algorithm detection: sha1, md5, md2, md4 (with RSA suffix variants)
- Case insensitivity: SHA1WithRSAEncryption correctly flagged
- Safe algorithms: sha256, sha384, sha512 return False
- Edge case: empty string, random string

## Notes

- Original test suite is fully preserved - this PR only adds to it
- Previous PR #1448 accidently replaced the existing tests; this PR corrects that by appending instead
- `is_weak_hash_algo()` coverage: 0% -> 15% on nettacker/core/lib/ssl.py

## Type of change

- [ ] New core framework functionality
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Code refactoring without any functionality changes
- [ ] New or existing module/payload change
- [ ] Documentation/localization improvement
- [x] Test coverage improvement
- [ ] Dependency upgrade
- [ ] Other improvement (best practice, cleanup, optimization, etc)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [ ] I have **digitally signed** all my commits in this PR
- [ ] I've run `make pre-commit` and confirm it didn't generate any warnings/changes
- [x] I've run `make test`, I confirm all tests passed locally
- [ ] I've added/updated any relevant documentation in the `docs/` folder 
- [ ] I've linked this PR with an open issue
- [x] I've tested and verified that my code works as intended and resolves the issue as described
- [ ] I have attached screenshots demonstrating my code works as intended
- [x] I've checked all other open PRs to avoid submitting duplicate work
- [x] I confirm that the code and comments in this PR are not direct unreviewed outputs of AI
- [x] I confirm that I am the Sole Responsible Author for every line of code, comment, and design decision

[contributing-guidelines]: https://nettacker.readthedocs.io/en/latest/Developers/

### CODE REVIEW

File: tests/core/lib/test_ssl.py [hunk 1]
```diff
@@ -572,3 +572,52 @@ def test_response_conditions_matched_none_response(self, ssl_engine, substeps):
...
+        assert is_weak_hash_algo("sha384WithRSAEncryption") is False
+
+    # --- Edge cases ---
+
+    def test_empty_string_does_not_crash(self):
+        # Empty string should return False, not raise an exception
+        assert is_weak_hash_algo("") is False
+
+    def test_random_string_is_not_weak(self):
+        assert is_weak_hash_algo("someRandomAlgorithm") is False
```
[the remark this answers is not in the document]
Comment (Raavi29) [reply tests/core/lib/test_ssl.py]: Issue resolved

**Comment by Raavi29** [comment]
Latest commit adds a test for weak algo not at start of string (rsaWithSHA1Encryption) - addresses CodeRabbit suggestion about non-prefix cases.

**Comment by securestep9** [comment]
@Raavi29 PRs unsigned commits will not be accepted ⚠️

**Comment by Raavi29** [comment]
> @Raavi29 PRs unsigned commits will not be accepted ⚠️

@securestep9 Duly noted.
I have resigned all the commits.
